client/templates/retry.tmpl

```diff
@@ -7,10 +7,8 @@ import (
 	"github.com/uber/cadence/common/types"
 )
 
-{{ $decorator := (printf "retryable%s" .Interface.Name) }}
-{{ $Decorator := (printf "Retryable%s" .Interface.Name) }}
-
-var _ {{.Interface.Name}} = (*{{$decorator}})(nil)
```

Comment:
possibly worth bringing back.
it's just a duplicated check done elsewhere, but it does help point out the specific issue, since it can sometimes be opaque which inferred type is referring to what and why it doesn't match at location X.

Reply:
There is a constructor that returns an interface that will point to the struct not matching the interface. So this is duplicate check, imo.